internal/pkg/service/stream/storage/level/local/encoding/config.go

```diff
@@ -0,0 +1,39 @@
+package encoding
+
+import (
+	"github.com/c2h5oh/datasize"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/encoding/compression"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/encoding/encoder"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/encoding/writesync"
+)
+
+// Config configures the local writer.
+type Config struct {
+	Encoder      encoder.Config     `configKey:"encoder"`
+	InputBuffer  datasize.ByteSize  `configKey:"inputBuffer" configUsage:"Max size of the buffer before compression, if compression is enabled. 0 = disabled" validate:"maxBytes=16MB"`
+	OutputBuffer datasize.ByteSize  `configKey:"outputBuffer" configUsage:"Max size of the buffer before the output. 0 = disabled" validate:"maxBytes=16MB"`
+	Compression  compression.Config `configKey:"compression"`
+	// SyncerFactory creates writesync.Syncer.
+	// A custom implementation can be useful for tests.
+	SyncerFactory writesync.SyncerFactory
+}
+
```

Comment:
The configuration needed some minor changes to match the logic/structure of the individual parts of the application.